vispy/scene/cameras/cameras.py

```diff
@@ -688,10 +689,42 @@ def _update_transform(self):
         # Apply mapping between viewbox and cam view
         self.transform.set_mapping(self._real_rect, vbr)
         # Scale z, so that the clipping planes are between -alot and +alot
-        self.transform.zoom((1, 1, 1/self._get_depth_value()))
+        self.transform.zoom((1, 1, 1/d))
+        
+        # We've now set self.transform, which represents our 2D
+        # transform When up is +z this is all. In other cases,
+        # self.transform is now set up correctly to allow pan/zoom, but
+        # for the scene we need a different (3D) mapping. When there
+        # is a minus in up, we simply look at the scene from the other
+        # side (as if z was flipped).
+        
+        if self.up == '+z':
+            # Keep it simple, but we give the scene an affinetransform
+            # to be consistent
+            thetransform = self.transform * AffineTransform()
```

Comment:
Doesn't it end up as a dot product on the GPU either way? Or do you mean in the scenegraph before it gets to the GPU?